app/recipe/serializers.py

```diff
@@ -0,0 +1,54 @@
+from rest_framework import serializers
+
+from core.models import Recipe, Ingredient
+
+
+class IngredientSerializer(serializers.ModelSerializer):
+    """Serializer for the Ingredient object"""
+
+    class Meta:
+        model = Ingredient
+        fields = ('id', 'name', )
+        read_only_fields = ('id', )
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    """Serializer for the Recipe object"""
+    ingredients = IngredientSerializer(
+        many=True,
+        required=False
+    )
+
+    class Meta:
+        model = Recipe
+        fields = ['id', 'name', 'description', 'ingredients']
+        read_only_fields = ('id', )
+
+    def create(self, validated_data):
+        """Create recipe related with the ingredients"""
+        ingredients = validated_data.pop('ingredients', [])
+        recipe = Recipe.objects.create(**validated_data)
+
+        for ingredient in ingredients:
+            Ingredient.objects.create(
+                name=ingredient['name'],
+                recipe=recipe
+            )
+
+        return recipe
+
+    def update(self, instance, validated_data):
+        """Update recipe related with the ingredients"""
+        ingredients = validated_data.pop('ingredients', None)
+        instance.ingredients.all().delete()
+
+        if ingredients is not None:
+            for ingredient in ingredients:
+                instance.ingredients.create(
+                    name=ingredient['name'],
+                    recipe=instance
+                )
+
+        super().update(instance, validated_data)
+        instance.save()
```

Comment:
Shouldn't this `save()` already be done within `super().update(instance, validated_data)`?

Reply:
I wasn't so sure that the update also commit all change, because I didn't find it in the documentation. If you say is like that I'll remove the save() invocation. Thanks :smiley:

Reply:
Yes, it's a bit hard to find. Check here: https://github.com/encode/django-rest-framework/blob/master/rest_framework/serializers.py#L991